components/server/src/model/source.py

```diff
@@ -0,0 +1,49 @@
+"""Source model class."""
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from model.metric import Metric
+
+
+class Source(dict):
+    """Class representing a measurement source."""
+
+    def __init__(self, metric: Metric, *args, **kwargs):
+        self.__metric = metric
+        super().__init__(*args, **kwargs)
+
+    def total(self) -> int:
+        """Return the measurement total of the source."""
+        return int(self["total"] or "100")
+
+    def value(self) -> int:
```

Comment:
What is the value (and the total) of a source? Isn't it the Metric (or a measurement) that has a value?

Reply:
A measurement can have multiple sources. The values of the sources are combined (by taking the sum, min or max) to arrive at the measurement value. Similar for the total.